### PR TITLE
Added per-path generator

### PR DESCRIPTION
**Pull request by wingedfox**
Added support for https://github.com/kogosoftwarellc/open-api/tree/master/packages/express-openapi

### CODE REVIEW

**Comment by MikeRalphson** [comment]
Build failure on node 10 appears unrelated to your changes (for which many thanks BTW).

**Comment by wingedfox** [comment]
It seems that Angular generator is broken or something like that.
I have no idea on what may be wrong with the javascript code which is almost identical to other generators... I'll try to look there.

**Comment by MikeRalphson** [comment]
The build failure is already fixed. It was due to a change in nodejs 10.

File: index.js [hunk 1]
```diff
@@ -123,7 +140,7 @@ function main(o, config, configName, callback) {
                             let cOperations = clone(api.operations);
                             let fnTemplate = Hogan.compile(po.output);
                             let template = Hogan.compile(ff.readFileSync(tpl(config, configName, po.input), 'utf8'));
-                            for (let operation of cOperations) {
+                            for (let operation of cOperations.operation) {
```
[the remark this answers is not in the document]
Comment (wingedfox) [reply index.js]: You're right, it is not directly related and perhaps it should go as another bugfix PR, but without this change `perOperation` generator does not work.

[the remark this answers is not in the document]
Comment (MikeRalphson) [reply index.js]: `perOperation` is not used by any of the currently-shipped config files.

File: adaptor.js [hunk 2]
```diff
@@ -482,6 +482,64 @@ function convertToApis(source,obj,defaults) {
...
+                    tagName = op.tags[0];
+                }
+                let entry = paths.find(function(e,i,a){
+                    return (e.name === p);
+                });
+                if (!entry) {
+                    const split = p.replace(/^\//,'').split(/\//g);
+                    const dirname = split.slice(0,-1).join('/');
+                    const filename = split.slice(-1).join('');
+                    const className = split.map(v=>v.replace(/{([^}]+)}/g,(v,v1)=>`By${v1[0].toUpperCase()}${v1.slice(1)}`).replace(/^./,(v)=>`${v[0].toUpperCase()}${v.slice(1)}`)).join('')
```
Comment (MikeRalphson) [inline adaptor.js]: Can we have a comment on what this map / regex is doing? Is it dependent on the path value having a particular format? Also missing semi-colon.

Comment (wingedfox) [reply adaptor.js]: Added comments

**Pull request merged**